dot/node.go

```diff
@@ -354,6 +391,34 @@ func setupMetricsServer(address string) {
 	}()
 }
 
+// stores the global node name to reuse
+func storeGlobalNodeName(name, basepath string) error {
+	db, err := state.SetupDatabase(basepath)
+	if err != nil {
+		return err
+	}
+
+	basestate := state.NewBaseState(db)
+	err = basestate.StoreNodeGlobalName(name)
+	if err != nil {
+		logger.Warn(
+			"failed to store global node name",
+			"basepath", basepath,
+			"error", err,
+		)
+		return nil
```

Comment:
why does this return `nil`? Shouldn't the caller of this method be aware if the global node name was stored?